consensus/babe/types.go

```diff
@@ -42,3 +48,68 @@ type AuthorityData struct {
 	id     *sr25519.PublicKey
 	weight uint64
 }
+
+var Timstap0 = []byte("timstap0")
+var Babeslot = []byte("babeslot")
+
+// InherentsData contains a mapping of inherent keys to values
+// Keys must be 8 bytes, values are a variable-length byte array
+type InherentsData struct {
+	data map[[8]byte]([]byte)
+}
+
+func NewInherentsData() *InherentsData {
+	return &InherentsData{
+		data: make(map[[8]byte]([]byte)),
+	}
+}
+
+func (d *InherentsData) SetInt64Inherent(key []byte, data uint64) error {
+	if len(key) != 8 {
+		return errors.New("inherent key must be 8 bytes")
+	}
+
+	val := make([]byte, 8)
+	binary.LittleEndian.PutUint64(val, data)
+
+	venc, err := scale.Encode(val)
+	if err != nil {
+		return err
+	}
+
+	kb := [8]byte{}
+	copy(kb[:], key)
+
+	d.data[kb] = venc
+	return nil
+}
+
+func (d *InherentsData) Encode() ([]byte, error) {
+	length := big.NewInt(int64(len(d.data)))
+
+	buffer := bytes.Buffer{}
+	se := scale.Encoder{Writer: &buffer}
+
+	_, err := se.Encode(length)
+	if err != nil {
+		return nil, err
+	}
+
+	for k, v := range d.data {
+		_, err = buffer.Write(k[:])
+		if err != nil {
+			return nil, err
+		}
+		_, err = buffer.Write(v)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return buffer.Bytes(), nil
+}
+
+type Slot struct {
```

Comment:
Should this be at the top of the file? Also, a comment would be nice explaining `Slot`.

Reply:
I've moved it and added a comment